bytestring/jvm/src/ByteStringJvmExt.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io.bytestring
+
+import java.nio.charset.Charset
+
+/**
+ * Decodes the content of a byte string to a string using given [charset].
+ *
+ * @param charset the charset to decode data into a string.
+ */
+public fun ByteString.toString(charset: Charset): String = getBackingArrayReference().toString(charset)
```

Comment:
This one breaks the general rules of methods overloading: two overloads _usually_ differ in their input set, but have the same semantics.

This is not the case for `toString()` vs `toString(Charset)`

Reply:
@whyoleg @qwwdfsad 
Maybe `encodeToString` / `decodeFromString` then? And `encodeToUtf8String` / `decodeFromUtf8String` for methods from the common source set.

Reply:
I think, that we should use `decode` for conversion to `String` (same as in stdlib) as we a decoding bytes into characters, nor otherwise. In that case for `ByteString` creation it will be `ByteString.Companion.encodeFromString(String): ByteString`, and I don't really like the naming of `encode from`.

I do like stdlib-like naming more: `ByteString.decodeToString(): String` and `String.encodeToByteString(): ByteString` for UTF8 + overload with `Charset` on JVM (with possibility of multiplatform charsets in future) + may be overloads with indices (like we have in stdlib)
Though not sure, that adding extension to `String` is the best idea.

Reply:
Adopted the following naming:
- `ByteString.decodeToString(): String`
- `String.encodeToByteString(): ByteString`

Reply:
nice!
last minor thing: as we not use `utf8` in naming anymore here, may be it make sense to do the same renaming in `core` module for `readUtf8` and `writeUtf8` (with `readString(Charset)`/`writeString(Charset)` on JVM) and replace common functions naming to use `readString`/`writeString`?

Reply:
Sure, sounds reasonable. Thanks for the suggestion! I filed https://github.com/Kotlin/kotlinx-io/issues/155 for it.